src/Models/ShallowWaterModels/shallow_water_advection_operators.jl

```diff
@@ -2,38 +2,51 @@ using Oceananigans.Advection:
+    _advective_momentum_flux_Vv,
+    vertical_vorticity_U,
+    vertical_vorticity_V,
+    bernoulli_head_U,
+    bernoulli_head_V
 
 using Oceananigans.Grids: AbstractGrid
 using Oceananigans.Operators: Ax_qᶠᶜᶜ, Ay_qᶜᶠᶜ
 
+
 #####
 ##### Momentum flux operators
 #####
 
-@inline momentum_flux_huu(i, j, k, grid, advection, solution) =
+@inline momentum_flux_huu(i, j, k, grid, advection, solution, formulation) =
     @inbounds _advective_momentum_flux_Uu(i, j, k, grid, advection, solution.uh, solution.uh) / solution.h[i, j, k]
 
-@inline momentum_flux_hvu(i, j, k, grid, advection, solution) =
+@inline momentum_flux_hvu(i, j, k, grid, advection, solution, formulation) =
     @inbounds _advective_momentum_flux_Vu(i, j, k, grid, advection, solution.vh, solution.uh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
 
-@inline momentum_flux_huv(i, j, k, grid, advection, solution) =
+@inline momentum_flux_huv(i, j, k, grid, advection, solution, formulation) =
     @inbounds _advective_momentum_flux_Uv(i, j, k, grid, advection, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
 
-@inline momentum_flux_hvv(i, j, k, grid, advection, solution) =
+@inline momentum_flux_hvv(i, j, k, grid, advection, solution, formulation) =
     @inbounds _advective_momentum_flux_Vv(i, j, k, grid, advection, solution.vh, solution.vh) / solution.h[i, j, k]
 
 #####
 ##### Momentum flux divergence operators
 #####
 
-@inline div_hUu(i, j, k, grid, advection, solution) =
-    1 / Vᶠᶜᶜ(i, j, k, grid) * (δxᶠᵃᵃ(i, j, k, grid, momentum_flux_huu, advection, solution) +
-                               δyᵃᶜᵃ(i, j, k, grid, momentum_flux_hvu, advection, solution))
+@inline div_hUu(i, j, k, grid, advection, solution, ::VectorInvariantFormulation) = (
+    + vertical_vorticity_U(i, j, k, grid, advection, solution.uh, solution.vh)  # Vertical relative vorticity term
+    + bernoulli_head_U(i, j, k, grid, advection, solution.uh, solution.vh))     # Bernoulli head term
+    
+@inline div_hUv(i, j, k, grid, advection, solution, ::VectorInvariantFormulation) = (
+    + vertical_vorticity_V(i, j, k, grid, advection, solution.uh, solution.vh)  # Vertical relative vorticity term
+    + bernoulli_head_V(i, j, k, grid, advection, solution.uh, solution.vh))     # Bernoulli head term
+
+@inline div_hUu(i, j, k, grid, advection, solution, formulation) =
+    1 / Vᶠᶜᶜ(i, j, k, grid) * (δxᶠᵃᵃ(i, j, k, grid, momentum_flux_huu, advection, solution, formulation) +
+                               δyᵃᶜᵃ(i, j, k, grid, momentum_flux_hvu, advection, solution, formulation))
 
-@inline div_hUv(i, j, k, grid, advection, solution) =
-    1 / Vᶜᶠᶜ(i, j, k, grid) * (δxᶜᵃᵃ(i, j, k, grid, momentum_flux_huv, advection, solution) +
-                               δyᵃᶠᵃ(i, j, k, grid, momentum_flux_hvv, advection, solution))
+@inline div_hUv(i, j, k, grid, advection, solution, formulation) =
+    1 / Vᶜᶠᶜ(i, j, k, grid) * (δxᶜᵃᵃ(i, j, k, grid, momentum_flux_huv, advection, solution, formulation) +
```

Comment:
This is what the code did before, and does work, but since we are really doing an area integral, would we want to have `Az` instead of `V`?  I recall that we pick `\Delta z = 1`, so they are equivalent but to me areas makes more sense than volumes, in this context.